yo1k/tic_tac_toe/test/core/test_game.py

```diff
@@ -0,0 +1,266 @@
+        self.assertEqual(state.phase, g.Phase.INROUND)
+        self.assertEqual(state.players[0].wins, 0)
+        self.assertEqual(state.players[1].wins, 0)
+
+    def test_advance_surrender_action(self):
+        state = _new_state()
+        act_queue_px = ListActionQueue([g.Action.new_surrender()])
+        g.Logic([act_queue_px, []]).advance(state)
+        next_game_state_board = [
+            [None, None, None], [None, None, None], [None, None, None]]
+        self.assertEqual(state.board.cells, next_game_state_board)
+        self.assertEqual(state.step, 0)
+        self.assertEqual(state.round, 0)
+        self.assertEqual(state.phase, g.Phase.OUTROUND)
+        self.assertEqual(state.players[0].wins, 0)
+        self.assertEqual(state.players[1].wins, 1)
+
+    def test_win_condition(self):  # SKTODO rename lists and try to pack them
+        list_of_cell_and_win_board = [
+            (g.Cell(0, 0),
+             [[g.Mark.X, g.Mark.X, g.Mark.X], [None, None, None], [None, None, None]]),
+            (g.Cell(1, 0),
+             [[None, None, None], [g.Mark.X, g.Mark.X, g.Mark.X], [None, None, None]]),
+            (g.Cell(2, 0),
+             [[None, None, None], [None, None, None], [g.Mark.X, g.Mark.X, g.Mark.X]]),
+            (g.Cell(1, 0),
+             [[g.Mark.X, None, None], [g.Mark.X, None, None], [g.Mark.X, None, None]]),
+            (g.Cell(1, 1),
+             [[None, g.Mark.X, None], [None, g.Mark.X, None], [None, g.Mark.X, None]]),
+            (g.Cell(1, 2),
+             [[None, None, g.Mark.X], [None, None, g.Mark.X], [None, None, g.Mark.X]]),
+            (g.Cell(1, 1),
+             [[g.Mark.X, None, None], [None, g.Mark.X, None], [None, None, g.Mark.X]]),
+            (g.Cell(1, 1),
+             [[None, None, g.Mark.X], [None, g.Mark.X, None], [g.Mark.X, None, None]])]
+        for i in list_of_cell_and_win_board:
+            cell, win_board = i
+            state = _new_state(board=g.Board(win_board))
+            with self.subTest(win_board=state.board.cells):
+                self.assertTrue(g.Logic._win_condition(state.board, cell))
+
+    def test_not_win_condition(self):
+        list_of_cell_and_not_win_board = [
+            (g.Cell(1, 0),
+             [[g.Mark.X, g.Mark.X, None], [g.Mark.X, None, None], [None, None, None]]),
+            (g.Cell(0, 0),
+             [[g.Mark.X, None, None], [None, None, g.Mark.X], [None, g.Mark.X, None]]),
+            (g.Cell(0, 1),
+             [[None, g.Mark.X, None], [None, None, g.Mark.X], [None, g.Mark.X, None]]),
+            (g.Cell(0, 2),
+             [[g.Mark.X, g.Mark.O, g.Mark.X], [None, None, None], [None, None, None]])]
+        for i in list_of_cell_and_not_win_board:
+            cell, not_win_board = i
+            state = _new_state(board=g.Board(not_win_board))
+            with self.subTest(not_win_board=state.board.cells):
+                self.assertFalse(g.Logic._win_condition(state.board, cell))
+
+    def test_advance_win_action(self):
```

Comment:
Something is really wrong with this test name, there is no such thing as `win_action`. The same is applicable to the test named `test_advance_draw_action`.

Reply:
Done.